src/accommodation/accommodation.controller.ts

```diff
@@ -288,6 +289,45 @@ export class AccommodationController {
     return { success: true, data: accommodations };
   }
 
+  @ApiOperation({ summary: 'Get all reviews to this accommodation' })
+  @ApiResponse({
+    status: 200,
+    description: 'Reviews of this accommodation',
+    schema: {
+      properties: {
+        success: { type: 'boolean' },
+        data: {
+          type: 'array',
+          items: {
+            $ref: getSchemaPath(ReviewDto),
+          },
+        },
+        countByRating: {
+          type: 'object',
+          additionalProperties: {
+            type: 'integer',
+          },
+        },
+        averageRate: {
+          type: 'number',
+        },
+      },
+    },
+  })
+  @ApiResponse({
+    status: 500,
+    description: 'Internal Server Error',
+  })
+  @ApiUnauthorizedResponse({
+    status: 401,
+    description: 'Unauthorized',
+  })
+  @Get('/:accommodationId/reviews')
+  async getAllReviews(@Param('accommodationId') accommodationId: string) {
+    const review = await this.accommodationService.getAccommodationReviews(accommodationId);
+    return { success: true, ...review };
```

Comment:
looks like spread unnecessary. review object available only inside function

Reply:
i need it bacause, from service I return all reviews in the data filed, and avarageRate and countingOfRating, there both shouldn't be inside the data field.
```js
return { data: reviews, countByRating, averageRate };
```

the final object to front need to look like this

```js
{
  "success": true,
  "data": [
    {
      "id": "string",
      "userId": "string",
      "accommodationId": "string",
      "feedback": "string",
      "rating": 0,
      "createdAt": "string",
      "user": {
        "id": "string",
        "firstName": "string",
        "lastName": "string",
        "profile": {
          "country": "string"
        }
      }
    }
  ],
  "countByRating": {
    "additionalProp1": 0,
    "additionalProp2": 0,
    "additionalProp3": 0
  },
  "averageRate": 0
}
```